src/framework-ondata.js

```diff
@@ -456,3 +459,51 @@ function shouldRunUpdater(vni, isStale) {
     // TODO: May add different policies in the future.
     return (!isStale) && haveAllInputs(vni) && dataIsGood(vni);
 }
+
+/**
+ * Set default output metadata by appending input state metadata to the outputState 
+ * so it gets passed on to downstream nodes.
+ */
+function setDefaultMetadata(vni, inputState) {
+
+    var inputKeys = Object.keys(inputState);
+    var outputState = vni.outputState();
+    var outputKeys = Object.keys(outputState);
+
+    // Walk to list of keys looking for non-standard state keys - those will be metadata.
+    inputKeys.forEach(function(key) {
+        if (!_.contains(createState.STATE_KEYS, key)) {
+
+            // Found a metadata key - do we already know about it on the output? 
+            if (_.contains(outputKeys, key)) { 
+
+                // We already have an entry for this metadata key.  
+                var metadata = outputState[key];
+                if (_.isArray(metadata) && !_.contains(metadata, inputState[key])) {
+                    // Metadata is an array - push this value onto it with any earlier values
+                    // So downstream node can see them all
+                    logger.debug('Framework found multiple metadata values for ',key);
+                    metadata.push(inputState[key]);
+                    metadata = _.uniq(metadata);
+
+                } else if ((!_.isArray(metadata)) && (metadata !== inputState[key])) {
+                    // Metadata is currently a single value and we have two different values for it
+
+                    // Is either the new value or the old value undefined?  If so, keep whatever the new value is
+                    if (_.isUndefined(inputState[key]) || _.isUndefined(metadata)) { 
```

Comment:
This should just check metadata, to avoid the situation when the first input has a value and a second input overrides the output value with it's undefined value.